ush/global_ens/evs_global_ens_atmos_grid2grid.sh

```diff
@@ -202,46 +197,82 @@ if [ $verify = upper ] ; then
               fi
           fi
         done
-        lead=$(echo $(echo ${lead_arr[@]}) | tr ' ' ',')
-        unset lead_arr
-        if [ $metplus_job = GridStat ] && [ $modnam = ecme ]; then
-          typeset -a lead_SFC_arr
-          for lead_chk in $leads_chk; do
-            fcst_time=$($NDATE -$lead_chk ${vday}${vhour})
-            fyyyymmdd=${fcst_time:0:8}
-            ihour=${fcst_time:8:2}
-            chk_file_SFC=${WORK}/grid2grid/run_${modnam}_valid_at_t${vhour}z_${fhr}/stat/${modnam}/GenEnsProd_${MODL}_g2g_BIN1_SFC_FHR${lead_chk}_${vday}_${vhour}0000V_ens.nc
-            if [ -s $chk_file_SFC ]; then
-              lead_SFC_arr[${#lead_SFC_arr[*]}+1]=${lead_chk}
+
+        if [ $modnam = naefs ] ; then
+	  conf_MODL="NAEFSbc"
+	elif [ $modnam = cmce ] ; then
+	  conf_MODL="GEFS"
+	else
+	  conf_MODL=$MODL
+	fi
+
+        for lead_hr in ${lead_arr[@]}; do
+          # Check for restart: check if the single sub-task is completed in the previous run
+	  # If this task has been completed in the previous run, then skip it
+	  if [ ! -e $COMOUTsmall/run_${modnam}_valid_at_t${vhour}z_${fhr}_${lead_hr}_${metplus_job}_g2g.completed ] ; then
+	    echo "export lead=${lead_hr}" >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
+
+            if [ $metplus_job = GenEnsProd ] || [ $metplus_job = EnsembleStat ]; then
+       	      echo "${METPLUS_PATH}/ush/run_metplus.py -c ${PARMevs}/metplus_config/machine.conf -c ${GRID2GRID_CONF}/${metplus_job}_fcst${conf_MODL}_obsModelAnalysis_climoERA5.conf " >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
+              echo "export err=\$?; err_chk" >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
+              if [ $modnam = ecme ]; then
+                echo "${METPLUS_PATH}/ush/run_metplus.py -c ${PARMevs}/metplus_config/machine.conf -c ${GRID2GRID_CONF}/${metplus_job}_fcst${conf_MODL}_obsModelAnalysis_climoERA5_SFC.conf " >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
+                echo "export err=\$?; err_chk" >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
+              fi
+            elif [ $metplus_job = GridStat ]; then
+              echo "${METPLUS_PATH}/ush/run_metplus.py -c ${PARMevs}/metplus_config/machine.conf -c ${GRID2GRID_CONF}/${metplus_job}_fcst${conf_MODL}_obsModelAnalysis_climoERA5_mean.conf " >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
+              echo "export err=\$?; err_chk" >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
             fi
+
+	    # Indicate sub-task is completed for restart 
+	    echo ">$COMOUTsmall/run_${modnam}_valid_at_t${vhour}z_${fhr}_${lead_hr}_${metplus_job}_g2g.completed" >> run_${modnam}_valid_at_t${vhour}z_${fhr}_${metplus_job}_g2g.sh
```

Comment:
It looks like this is writing directly to COMOUTsmall. We can only copy files there.

Can the restart file get written to DATA, and then `if [ $SENDCOM = YES ] ; then` copy to COMOUTsmall?

Reply:
Added.